geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -229,6 +261,32 @@ public int getDSFID() {
     return REDIS_LIST_ID;
   }
 
+  public synchronized int elementInsert(byte[] elementToInsert, byte[] referenceElement,
+      boolean before) {
+    int i = 0;
```

Comment:
Rather than using a separate counter here, you could instead use the `nextIndex()` or `previousIndex()` methods in `ListIterator` to return the appropriate index. Really the only reason to create a `ListIterator` rather than a plain `Iterator` here is to have access to those methods.

Reply:
for some reason this didn't work. I looked at it for awhile but don't see why it wouldn't. I left the code as is.